extendr-api/src/robj/into_robj.rs

```diff
@@ -39,16 +51,157 @@ impl From<()> for Robj {
+            }
+            //can only imagine this would ever fail due memory allcation error, but then panicking is the right choice
+            .expect("internal error: failed to create an R list")
+            .set_class(["extendr_error", "error", "condition"])
+            .expect("internal error: failed to set class"),
+        }
+    }
+}
+
+/// Convert a Result to an R `List` with an `ok` and `err` element.
+/// This is used to allow functions to use the ? operator
+/// and return [std::result::Result<T,E> or extendr_api::result::Result<T>]
+/// without panicking on an Err.
+///
+///
+/// ```
+/// use extendr_api::prelude::*;
+/// fn my_err_f() -> std::result::Result<f64, String> {
+///     Err("We have water in the engine room!".to_string())
+/// }
+/// fn my_ok_f() -> std::result::Result<f64, String> {
+///     Ok(123.123)
+/// }
+///
+/// test! {
+///     assert_eq!(
+///         r!(my_err_f()),
+///         R!("x=list(ok=NULL, err='We have water in the engine room!')
+///             class(x)='extendr_result'
+///             x"
+///         ).unwrap()
+///     );
+///     assert_eq!(
+///         r!(my_ok_f()),
+///         R!("x = list(ok=123.123, err=NULL)
+///             class(x)='extendr_result'
+///             x"
+///         ).unwrap()
+///     );
+/// }
+///
+/// ```
+#[cfg(feature = "result_list")] //write this feature and not any result_feature with presecence
+impl<T, E> From<std::result::Result<T, E>> for Robj
+where
+    T: Into<Robj>,
+    E: Into<Robj>,
+{
+    fn from(res: std::result::Result<T, E>) -> Self {
+        match res {
+            Ok(x) => List::from_names_and_values(&["ok", "err"], &[x.into(), NULL.into()]),
+            Err(x) => {
+                let err_robj = x.into();
+                if err_robj.is_null() {
+                    panic!("Internal error: result_list not allowed to return NULL as err-value")
+                }
+                List::from_names_and_values(&["ok", "err"], &[NULL.into(), err_robj])
+            }
+        }
+        //can only imagine this would ever fail due memory allcation error, but then panicking is the right choice
+        .expect("Internal error: failed to create an R list")
+        .set_class(&["extendr_result"])
+        .expect("Internal error: failed to set class")
+        .into()
+    }
+}
+
+// string conversions from Error trait to Robj and String
+impl From<Error> for Robj {
+    fn from(res: Error) -> Self {
+        res.to_string().into()
+    }
+}
+impl From<Error> for String {
+    fn from(res: Error) -> Self {
+        res.to_string().into()
+    }
+}
+
+// // ... and Err does not have to implement Display
+// impl<T, E> From<std::result::Result<T, E>> for Robj
+// where
+//     T: Into<Robj>,
+//     E: Into<Robj>,
+// {
+//     fn from(res: std::result::Result<T, E>) -> Self {
+//         match res {
+//             Ok(x) => x.into(),
+//             Err(x) => x.into_robj().set_attrib("extendr_err", true).unwrap(),
+//         }
+//     }
+// }
```

Comment:
Is this code needed here? If you do not plan to uncomment it, it should be removed.

Reply:
removed